secret_store/src/key_server_cluster/cluster_connections_net.rs

```diff
@@ -0,0 +1,543 @@
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::collections::btree_map::Entry;
+use std::io;
+use std::net::{SocketAddr, IpAddr};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use futures::{future, Future, Stream};
+use parking_lot::{Mutex, RwLock};
+use tokio::net::{TcpListener, TcpStream};
+use tokio::timer::{Interval, timeout::Error as TimeoutError};
+use tokio_io::IoFuture;
+use ethkey::KeyPair;
+use parity_runtime::Executor;
+use key_server_cluster::{Error, NodeId, ClusterConfiguration, NodeKeyPair};
+use key_server_cluster::cluster_connections::{ConnectionProvider, Connection, ConnectionManager};
+use key_server_cluster::connection_trigger::{Maintain, ConnectionTrigger};
+use key_server_cluster::cluster_message_processor::MessageProcessor;
+use key_server_cluster::io::{DeadlineStatus, ReadMessage, SharedTcpStream,
+	read_encrypted_message, WriteMessage, write_encrypted_message};
+use key_server_cluster::message::{self, ClusterMessage, Message};
+use key_server_cluster::net::{accept_connection as io_accept_connection,
+	connect as io_connect, Connection as IoConnection};
+
+/// Empty future.
+pub type BoxedEmptyFuture = Box<Future<Item = (), Error = ()> + Send>;
+
+/// Maintain interval (seconds). Every MAINTAIN_INTERVAL seconds node:
+/// 1) checks if connected nodes are responding to KeepAlive messages
+/// 2) tries to connect to disconnected nodes
+/// 3) checks if enc/dec sessions are time-outed
+const MAINTAIN_INTERVAL: u64 = 10;
+
+/// When no messages have been received from node within KEEP_ALIVE_SEND_INTERVAL seconds,
+/// we must send KeepAlive message to the node to check if it still responds to messages.
+const KEEP_ALIVE_SEND_INTERVAL: Duration = Duration::from_secs(30);
+/// When no messages have been received from node within KEEP_ALIVE_DISCONNECT_INTERVAL seconds,
+/// we must treat this node as non-responding && disconnect from it.
+const KEEP_ALIVE_DISCONNECT_INTERVAL: Duration = Duration::from_secs(60);
+
+/// Network connection manager configuration.
+pub struct NetConnectionsManagerConfig {
+	/// Allow connecting to 'higher' nodes.
+	pub allow_connecting_to_higher_nodes: bool,
+	/// Interface to listen to.
+	pub listen_address: (String, u16),
+	/// Should key servers set change session when servers set changes? This
```

Comment:
Can you rephrase the first sentence? I don't understand it and also both `key servers` and `servers` are used here do they mean the same thing in this context?

I suggest adding a `TODO` here if the question is not sorted out